waiter/integration/waiter/token_request_test.clj

```diff
@@ -1429,6 +1431,8 @@
           (let [response-1 (make-request-with-debug-info request-headers #(make-request waiter-url "/hello" :headers %))]
             (assert-response-status response-1 200)
             (is (= service-id-1 (:service-id response-1))))
+          ;; allow every router to learn about the new instance
+          (Thread/sleep scheduler-sync-allowance-sleep-duration-ms)
```

Comment:
Why not use the wait-for pattern?

Reply:
Done, ready for another round of review.